server/src/service/job-service.ts

```diff
@@ -48,6 +49,12 @@ export class JobService {
   ): Promise<string> {
     const id = uuidv4()
     const tempInputPath = await getTempInputPath(id)
+
+    if (config.uploadUrl) {
+      // Check if uploadUrl is supported
+      this.getUploadService(config.uploadUrl)
+    }
```

Comment:
Add a "check" here so that "Unsupported URL error" could be thrown to the client before it's added to the job queue.